Tests/BusinessLogic.Tests/UnitTests/LogicTests/PlayedGamesTests/PlayedGameRetrieverTests/SearchPlayedGamesTests.cs

```diff
@@ -370,5 +372,32 @@ public void ItFiltersOnTheInclusionExternalSourceName()
             //--assert
             Assert.True(results.All(result => result.ApplicationLinkages.Any(x => x.ApplicationName == EXTERNAL_SOURCE_NAME)));
         }
+
+        [Test]
+        public void ItAddsPlayerIdToFilterIfWeOnlyWantCurrentPlayerGames()
+        {
+            ApplicationUser CurrentUser = new ApplicationUser()
+            {
+                Id = "1",
+                CurrentGamingGroupId = 1
+            };
+            Player CurrentPlayer = new Player()
+            {
+                Id = 1
+            };
+            PlayedGameFilter filter = new PlayedGameFilter()
+            {
+                IncludeCurrentPlayerGamesOnly = true
+            };
+            autoMocker.Get<IPlayerRetriever>().Expect(mock => mock.GetPlayerForCurrentUser(
+                    CurrentUser.Id, CurrentUser.CurrentGamingGroupId))
+                .Return(CurrentPlayer);
+
+            autoMocker = new RhinoAutoMocker<PlayedGameRetriever>();
+            autoMocker.Get<IDataContext>().Expect(mock => mock.GetQueryable<PlayedGame>()).Return(new List<PlayedGame>().AsQueryable());
+            autoMocker.ClassUnderTest.SearchPlayedGames(filter);
+
+            Assert.That(filter.PlayerId, Is.EqualTo(1));
```

Comment:
I'm making the assumption that the filter is permanently updated, which may not be the case. I'm considering doing something like
```
Arg<PlayedGameFilter>.Matches(
                    x => x.PlayerId == CurrentPlayer.Id)
        .Return(expectedSearchResults);
```
like in `SearchHttpPostTests`. We still wouldn't really have an Assert in that case, so I'd need to figure something out